### PR TITLE
[MM-29523] EmailVerified flag can be user as parameter

### PR DESCRIPTION
**Pull request by ethervoid**
#### Summary

Now the `EmailVerified` property in the user struct can be set and used by the admin to create users that are already verified

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-29523

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: api4/user_test.go [hunk 1]
```diff
@@ -24,11 +24,20 @@ func TestCreateUser(t *testing.T) {
...
+		Password:      "hello1",
+		Username:      GenerateTestUsername(),
+		Roles:         model.SYSTEM_ADMIN_ROLE_ID + " " + model.SYSTEM_USER_ROLE_ID,
+		EmailVerified: true,
+	}
 
 	ruser, resp := th.Client.CreateUser(&user)
 	CheckNoError(t, resp)
 	CheckCreatedStatus(t, resp)
+	// Creating user is not admin so we cannot set the email verified flag
```
Comment (isacikgoz) [inline api4/user_test.go:39]:
```suggestion
	// Creating a user as a regular user with verified flag should not verify the new user.
```

File: api4/user_test.go [hunk 2]
```diff
@@ -24,11 +24,20 @@ func TestCreateUser(t *testing.T) {
...
+	require.Equal(t, false, ruser.EmailVerified)
```
Comment (isacikgoz) [inline api4/user_test.go:40]: We can user `require.False` instead.
```suggestion
	require.False(t, ruser.EmailVerified)
```

File: api4/user_test.go [hunk 3]
```diff
@@ -67,9 +76,11 @@ func TestCreateUser(t *testing.T) {
 	th.App.UpdateConfig(func(cfg *model.Config) { *cfg.TeamSettings.EnableUserCreation = false })
 
 	th.TestForSystemAdminAndLocal(t, func(t *testing.T, client *model.Client4) {
-		user2 := &model.User{Email: th.GenerateTestEmail(), Password: "Password1", Username: GenerateTestUsername()}
-		_, resp = client.CreateUser(user2)
+		user2 := &model.User{Email: th.GenerateTestEmail(), Password: "Password1", Username: GenerateTestUsername(), EmailVerified: true}
+		ruser2, resp := client.CreateUser(user2)
 		CheckNoError(t, resp)
+		// Creating user is admin should allow set the email verified flag
```
Comment (isacikgoz) [inline api4/user_test.go:82]:
```suggestion
		// Creating a user as sysadmin should verify the user with the EmailVerified flag.
```

File: api4/user_test.go [hunk 4]
```diff
@@ -67,9 +76,11 @@ func TestCreateUser(t *testing.T) {
...
+		require.Equal(t, true, ruser2.EmailVerified)
```
Comment (isacikgoz) [inline api4/user_test.go:83]:
```suggestion
		require.True(t, ruser2.EmailVerified)
```

**Review by isacikgoz — APPROVED** [review]
LGTM, have some minor suggestions for readability.

**Review by mgdelacroix — APPROVED** [review]
LGTM! :+1:

**Comment by ethervoid** [comment]
@jupenur @DSchalla Hi! I know you're swamped but we need a quick review here so we can merge this for GA

**Comment by DSchalla** [comment]
For context, where do we plan to utilize this? Is this Cloud only? Is this for the system console?

**Comment by ethervoid** [comment]
@DSchalla We're using it from Cloud through mmctl so we can avoid the email verification for the first user that is created automatically by CWS.

**Review by DSchalla — APPROVED** [review]
Thanks @ethervoid, change itself looks good as long as we don't modify how any other creation flows work right now.

**Comment by ethervoid** [comment]
@DSchalla cool thx Daniel. Basically the change here is that we always set that to false unless the user is an admin so it should be secure

**Pull request merged**